src/web/app/pages-instructor/instructor-student-records-page/instructor-student-records-page.component.ts

```diff
@@ -56,10 +75,103 @@ export class InstructorStudentRecordsPageComponent implements OnInit {
    * Loads the student's records based on the given course ID and email.
    */
   loadStudentRecords(): void {
+    this.studentService.getStudent(this.courseId, this.studentEmail).subscribe((resp: Student) => {
+      this.studentSection = resp.sectionName;
+    });
     this.studentProfileService.getStudentProfile(this.studentEmail, this.courseId).subscribe((resp: StudentProfile) => {
       this.studentProfile = resp;
     }, (resp: ErrorMessageOutput) => {
       this.statusMessageService.showErrorMessage(resp.error.message);
     });
   }
+
+  /**
+   * Loads the student's feedback session results based on the given course ID and student name.
+   */
+  loadStudentResults(): void {
+    this.feedbackSessionsService.getFeedbackSessionsForInstructor(this.courseId).subscribe((resp: FeedbackSessions) => {
+      resp.feedbackSessions.forEach((feedbackSession: FeedbackSession) => {
+        this.feedbackSessionsService.getFeedbackSessionsResult({
+          courseId: this.courseId,
+          feedbackSessionName: feedbackSession.feedbackSessionName,
+          groupBySection: this.studentSection,
+          intent: Intent.INSTRUCTOR_RESULT,
+        }).subscribe((results: SessionResults) => {
+          results.questions.forEach((q: QuestionOutput) => {
+            q.allResponses = q.allResponses.filter((response: ResponseOutput) =>
```

Comment:
Here you will know whether the student is giver or recipient.

Try to separate them to two groups.

Reply:
I did try but I couldn't filter them properly so I reverted to single question set. I'll go back to the two groups and ask for help if I get stuck again :")

Reply:
You can just do make copy of `results` and filter them separately. In addition, please avoid name like `q`.